linked_list/main.py

```diff
@@ -42,54 +42,103 @@ class Node(object):
     """
     Node class representing each of the linked nodes in the list.
     """
-
     def __init__(self, elem, next=None):
-        pass
+        self.elem = elem
+        self.next = next
 
     def __str__(self):
-        pass
+        return str(self.elem)
 
     def __eq__(self, other):
-        pass
+        return isinstance(other, Node) and self.elem == other.elem
 
     def __repr__(self):
-        pass
+        return "Node: {0} with next node: {1}".format(self.elem, self.next)
 
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+            
+        if elements:
+            for element in elements:
+                self.append(element)
 
     def __str__(self):
-        pass
+        return '[{}]'.format(', '.join(str(i) for i in self))
 
     def __len__(self):
-        pass
+        return sum(1 for i in self)
 
     def __iter__(self):
-        pass
+        current = self.start
+        while current:
+            yield current
+            current = current.next
 
     def __getitem__(self, index):
-        pass
+        for i, node in enumerate(self):
+            if i == index:
+                return node.elem
 
     def __add__(self, other):
-        pass
+        new_list = []
```

Comment:
I thought other was passed in as a a list anyway, so I'm not sure if making a new list was necessary. I could be wrong though